### ISSUE TITLE
Inline schema without name gets a non human friendly name

### ISSUE THREAD

**Issue by joheredi** [Azure/autorest#4358]
For example this definition

https://github.com/Azure/azure-rest-api-specs/blob/a5f8ef67c8170e4081527e400473c6deddcfabfd/specification/search/data-plane/Azure.Search/preview/2021-04-30-Preview/searchservice.json#L359

Would end up producing a name like this: `Paths1Cj7DxmIndexersIndexernameSearchResetdocsPostRequestbodyContentApplicationJsonSchema`

This is, IMO a swagger issue and the fix would be to add `x-ms-client-name`.

Is there a way we can validate and enforce that these inline schemas have a name?

**Comment by joheredi** [comment]
Something similar happens with Inline array definitions where the values are defined as nameless enums:

https://github.com/Azure/autorest.typescript/blob/f7448a0c800f7fd3b84e37e4e4e02fd85f737075/test/integration/swaggers/appconfiguration.json#L1056

We get Enum1, Enum2, Enum3, etc.

**Comment by timotheeguerin** [comment]
I agree that this is terrible and there is but this is the job of the azure-rest-api-specs linting. Autorest shouldn't fail on that.

**Comment by joheredi** [comment]
I think that's a good idea and we'd be leveraging TS representing these definitions inline. Other languages that can't do this would still have the problem, right? They'll need a swagger fix

@sarangan12 - FYI. I don't think adding support for this would be super urgent for TS codegen as other languages would still suffer from the calculated names. However, if we want to reduce the use of calculated names in TS packages we could choose to inline anonymous definitions
